src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs

```diff
@@ -0,0 +1,264 @@
+                            definitionsList.Add(JsonConvert.SerializeObject(processingRecord));
+                            sequence++;
+                        }
+                    }
+
+                    definitions = definitionsList.ToArray();
+                }
+
+                groupJobs = await _queueClient.EnqueueAsync((byte)QueueType.Export, definitions, _jobInfo.GroupId, false, false, cancellationToken);
+            }
+
+            bool allJobsComplete;
+            do
+            {
+                allJobsComplete = true;
+                foreach (var job in groupJobs)
+                {
+                    if (job.Id != _jobInfo.Id && (job.Status == JobStatus.Running || job.Status == JobStatus.Created))
+                    {
+                        allJobsComplete = false;
+                        break;
+                    }
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(10), cancellationToken);
+                groupJobs = await _queueClient.GetJobByGroupIdAsync((byte)QueueType.Export, _jobInfo.GroupId, false, cancellationToken);
+            }
+            while (!allJobsComplete);
+
+            bool jobFailed = false;
+            foreach (var job in groupJobs)
+            {
+                if (job.Id != _jobInfo.Id)
+                {
+                    if (!string.IsNullOrEmpty(job.Result) && !job.Result.Equals("null", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var processResult = JsonConvert.DeserializeObject<ExportJobRecord>(job.Result);
+                        foreach (var output in processResult.Output)
+                        {
+                            if (record.Output.TryGetValue(output.Key, out var exportFileInfos))
+                            {
+                                exportFileInfos.AddRange(output.Value);
+                            }
+                            else
+                            {
+                                record.Output.Add(output.Key, output.Value);
+                            }
+                        }
+
+                        if (processResult.FailureDetails != null)
+                        {
+                            if (record.FailureDetails == null)
+                            {
+                                record.FailureDetails = processResult.FailureDetails;
+                            }
+                            else if (!processResult.FailureDetails.FailureReason.Equals(record.FailureDetails.FailureReason, StringComparison.OrdinalIgnoreCase))
+                            {
+                                record.FailureDetails = new JobFailureDetails(record.FailureDetails.FailureReason + "\r\n" + processResult.FailureDetails.FailureReason, record.FailureDetails.FailureStatusCode);
+                            }
+
+                            jobFailed = true;
```

Comment:
I don't see a good way to factor this out since the values assigned to FailureDetails is different in each case. The if statements are also slightly different between the two cases.